### DISCUSSION TITLE
memory deallocation

### DISCUSSION THREAD

**Discussion by pradkrish** [h2oai/datatable#2692] [General]
After working with smart pointers for many years, one begins to forget worrying about freeing memory. Forgetting this is bad when suddenly one goes back to working with raw pointers (something I did in the latest PR). Good that It was spotted during the review, which is why there is a code review system. is there some kind of checker that can integrated into our build system that can flag any unallocated memory?

**Comment by st-pasha** [comment]
We use AddressSanitizer (see `make asan` in the Makefile), though making it run reliably is not exactly easy.
Then, there is also Valgrind, but it always report TONS of false positives from the python itself, so this step can't be easily automated either.

On the other hand, if you do know how to set up either of these so that they can be run automatically, then that would be a welcome addition.
